tools/fastq_provider/fastq_provider.xml

```diff
@@ -1,78 +1,160 @@
+        <conditional name="fetch_type">
+            <param name="script" argument="-a" type="select" label="Use file/URI or library" help="Download using a path (file or URI) or library ID." refresh_on_change="true" optional="false">
+                <option value="fetchFastq">File or URI</option>
+                <option value="fetchEnaLibraryFastqs" selected="true">Library</option>
+            </param>
+            <when value="fetchFastq">
+                <param name="path" argument="-f" type="text" label="File or URI" help="Use a file path or URI." optional="false" />
+                <param name="target" argument="-t" type="text" label="Target file" help="Provide the target file name, e.g. 'ERR035229.fastq.gz'" optional="false" />
+                <param name="validate_only" argument="-v" type="boolean" label="Validate only, don't download" help="Check if you don't want to download and you only want to validate that a source URI is valid." checked="false" optional="true"/>
+                <param name="library" argument="-l" type="text" label="ENA library (ENA source files only), by default inferred from file name" help="The library ID, which, by default, is inferred from file name. E.g. -l ERR1888646" optional="true"/>
+            </when>
+            <when value="fetchEnaLibraryFastqs">
+                <param name="library" argument="-l" type="text" label="ENA library (ENA source files only), by default inferred from file name" help="The library ID, which, by default, is inferred from file name. E.g. -l ERR1888646" optional="false"/>
+                <!-- Below possibly not needed if added to the command -->
+                <!-- <param name="output_dir" argument="-d" type="text" label="Output directory" help="Where files are saved" optional="false" /> -->
+                <param name="download_type" argument="-t" type="select" label="download type, fastq or SRA file" help="download type, fastq or srr" optional="true">
+                    <option value="fastq" selected="true">fastq</option>
+                    <option value="srr">SRA</option>
+                </param>
+                <param name="sepe" argument="-n" type="select" label="SINGLE-end or PAIRED-end" help="SINGLE-end or PAIRED-end, default PAIRED" optional="true">
+                    <option value="PAIRED" selected="true">PAIRED</option>
+                    <option value="SINGLE">SINGLE</option>
+                </param>
+            </when>
+        </conditional>
+        <param name="retrieval_method" argument="-m" type="select" label="Retrieval method" help="Retrieval method, default 'auto'. Files with HCA source will use the 'hca' download method regardless (-m will have no effect). Private ENA files will be fetched via 'ssh' (-m will have has no effect). The 'http' method refers to an HTTP endpoint in the EBI Fire resource." optional="true">
```

Comment:
```Retrieval method``` should be removed/disabled depending on ```Resource or directory```

Reply:
> @pmb59 and @irisdianauy can you please add couple of tests utilising `Resources and directory` and `Retrieval Method` options.

Done in 7e6a086adab03a9d2d385a7c8a4df720b017f546.  I did not add an explicit `retrieval_method` parameter in the tests because the remaining options (`http`, `ftp`, and `ssh`) are for EBI only, and they will fail in the CI test.  The `ssh` will need keys/a password.  Please correct me if I'm wrong, Anil and/or @pmb59.  I have also removed `wget` as an option that can be explicitly set by the user, because doing so gives a tool error (see [newly opened issue](https://github.com/ebi-gene-expression-group/atlas-fastq-provider/issues/20)).

Reply:
As for the `retrieval_method` being disabled depending on the selected `resource`, I didn't include the logic for this in the wrapper because the tool has its own way of handling it.  e.g., `resource="hca"` overrides the `retrieval_method`.

Reply:
The `ssh` retrieval method has passed now (tested in EBI cluster), but the test cannot be included here becuase identity files are private.

Reply:
Created issue #267 so Retrieval methods can be looked at in future..